src/backend/cuda/jit/BufferNode.hpp

```diff
@@ -16,4 +16,20 @@ namespace jit {
 template<typename T>
 using BufferNode = common::BufferNodeBase<std::shared_ptr<T>, Param<T>>;
 }
+
 }  // namespace cuda
+
+namespace common {
+
+template<typename DataType, typename ParamType>
+bool BufferNodeBase<DataType, ParamType>::operator==(
+    const BufferNodeBase<DataType, ParamType> &other) const noexcept {
+    if (m_data.get() == other.m_data.get()) {
+        if (m_bytes == other.m_bytes) {
+            if (m_param.ptr == other.m_param.ptr) { return true; }
+        }
+    }
+    return false;
```

Comment:
```suggestion
    return m_data.get() == other.m_data.get() &&
           m_bytes == other.m_bytes &&
           m_param.offset == other.m_param.offset;
```
I think this is easier to read and less lines